notebooks/1-sequence-stats.jmd

```diff
@@ -0,0 +1,147 @@
+using Plots: make_fillrange_from_ribbon
+# Coronavirus Analysis
+Working directory must be Cov2_Analysis directory
+
+## Mean and Standard Deviation
+Finding mean sequence length, standard devation for length, mean gc content and standard deviation for gc content for CoV sequences.
+
+```julia 
+using BISC195Bioinformatics
+
+return mean_and_std_fasta("data/CoV_sequences.fasta")
+```
+
+## Minimum and Maximum
+
+Finding length of smallest and largest sequence for CoV sequences. 
+
+```julia
+seq_lengths = sequence_lengths("data/CoV_sequences.fasta")
+
+min_length = minimum(seq_lengths)
+max_length = maximum(seq_lengths)
+
+return min_length, max_length
+```
+
+## Identifying Outliers
+
+Using histogram to spot sequence length outliers in CoV sequences. Added more descriptive axis labels and removed legend.
+
+```julia
+using Plots
+
+histogram(seq_lengths, 
+xlabel = "Sequence Length",
+ylabel = "Frequency",
+legend = false)
+```
+## Filtering sequences with more than 25k bases
+
+Modifying sequence_length function from BISC195 Bioinformatics package to output new vectors only containing sequences with >25k bases.
+(Also creates new vector with adjoining headers).
+
+I modified the sequence_length function instead of just calling it because sequence_length already pulls the header and sequence arrays from parse_fasta. 
+This way I don't have to call parse_fasta a second time to get the original vectors for a slice. 
+Included sequence lengths in tuple output for use in next step.
+
+```julia
+"""
+    morethan25k(path)
+
+Given a path to a fasta file, outputs tuple containing only sequences, headers, and sequence lengths for sequences with more than 25k bases.
+"""
+function morethan25k(path)
+    seq_lengths = []
+    headers = parse_fasta(path)[1]
+    sequences = parse_fasta(path)[2]
+
+    for sequence in sequences 
```

Comment:
Instead of a loop, you could also do `seq_lengths = [length(sequence) for sequence in sequences]` - this is called an "array comprehension."